langchain/memory/chat_message_histories/zep.py

```diff
@@ -0,0 +1,142 @@
+    HumanMessage,
+)
+
+if TYPE_CHECKING:
+    from zep_python import Memory, Message, NotFoundError, SearchResult
+
+logger = logging.getLogger(__name__)
+
+
+class ZepChatMessageHistory(BaseChatMessageHistory):
+    """A ChatMessageHistory implementation that uses Zep as a backend.
+
+    Zep provides long-term conversation storage for LLM apps. The server stores,
+    summarizes, embeds, indexes, and enriches conversational AI chat
+    histories, and exposes them via simple, low-latency APIs.
+
+    For server installation instructions and more, see: https://getzep.github.io/
+
+    This class is a thin wrapper around the zep-python package. Additional
+    Zep functionality is exposed via the `zep_summary` and `zep_messages`
+    properties.
+
+    For more information on the zep-python package, see:
+    https://github.com/getzep/zep-python
+    """
+
+    def __init__(
+        self,
+        session_id: str,
+        url: str = "http://localhost:8000",
+    ) -> None:
+        try:
+            from zep_python import ZepClient
+        except ImportError:
+            raise ValueError(
+                "Could not import zep-python package. "
+                "Please install it with `pip install zep-python`."
+            )
+
+        self.zep_client = ZepClient(base_url=url)
+        self.session_id = session_id
+
+    @property
+    def messages(self) -> List[BaseMessage]:  # type: ignore
+        """Retrieve messages from Zep memory"""
+        zep_memory: Optional[Memory] = self._get_memory()
+        if not zep_memory:
+            return []
+
+        messages: List[BaseMessage] = []
+        # Extract summary, if present, and messages
+        if zep_memory.summary:
+            if len(zep_memory.summary.content) > 0:
+                messages.append(HumanMessage(content=zep_memory.summary.content))
+        if zep_memory.messages:
+            msg: Message
+            for msg in zep_memory.messages:
+                if msg.role == "ai":
+                    messages.append(AIMessage(content=msg.content))
+                else:
+                    messages.append(HumanMessage(content=msg.content))
+
+        return messages
+
+    @property
+    def zep_messages(self) -> List[Message]:
+        """Retrieve summary from Zep memory"""
+        zep_memory: Optional[Memory] = self._get_memory()
+        if not zep_memory:
+            return []
+
+        return zep_memory.messages
+
+    @property
+    def zep_summary(self) -> Optional[str]:
+        """Retrieve summary from Zep memory"""
+        zep_memory: Optional[Memory] = self._get_memory()
+        if not zep_memory or not zep_memory.summary:
+            return None
+
+        return zep_memory.summary.content
+
+    def _get_memory(self) -> Optional[Memory]:
+        """Retrieve memory from Zep"""
+        from zep_python import NotFoundError
+
+        try:
+            zep_memory: Memory = self.zep_client.get_memory(self.session_id)
+        except NotFoundError:
```

Comment:
would we maybe just want to error here? guessing this means session id is invalid and user should fix?

Reply:
if above is true this could also be a validation check to run at instantiation

Reply:
Zep allows developers to create sessions on demand when a memory is added. It's plausible that `_get_memory` is called before a session is created i.e. if the `messages` property is accessed prior to any messages having been added.